build/makelib/golang.mk

```diff
@@ -174,26 +176,26 @@ go.vet:
 go.fmt:
 	@$(GOHOST) fmt $(GO_ALL_PACKAGES)
 
+go.validate: go.vet go.fmt
+
 .PHONY: go.vendor
-go.vendor $(GO_VENDOR_DIR)/vendor.stamp: $(GO_TOOLS_DIR)/glide
+go.vendor $(GO_VENDOR_DIR)/vendor.stamp: $(GLIDE)
 	@mkdir -p $(GLIDE_HOME)
 	@$(GLIDE) install
 	@touch $(GO_VENDOR_DIR)/vendor.stamp
 
-$(GO_TOOLS_DIR)/glide:
+$(GLIDE):
 	@echo "installing glide"
-	@mkdir -p $(GO_TOOLS_DIR)
-	@curl -sL https://github.com/Masterminds/glide/releases/download/v0.12.3/glide-v0.12.3-$(GOHOSTOS)-$(GOHOSTARCH).tar.gz | tar -xz -C $(GO_TOOLS_DIR)
-	@mv $(GO_TOOLS_DIR)/$(GOHOSTOS)-$(GOHOSTARCH)/glide $(GO_TOOLS_DIR)/glide
-	@rm -r $(GO_TOOLS_DIR)/$(GOHOSTOS)-$(GOHOSTARCH)
+	@mkdir -p $(GO_TOOLS_HOST_DIR)/tmp
+	@curl -sL https://github.com/Masterminds/glide/releases/download/v0.12.3/glide-v0.12.3-$(GOHOSTOS)-$(GOHOSTARCH).tar.gz | tar -xz -C $(GO_TOOLS_HOST_DIR)/tmp
+	@mv $(GO_TOOLS_HOST_DIR)/tmp/$(GOHOSTOS)-$(GOHOSTARCH)/glide $(GO_TOOLS_HOST_DIR)/glide
+	@rm -r $(GO_TOOLS_HOST_DIR)/tmp
```

Comment:
there's no easy way to do this in a makefile. I dont think its worth handling. the next run should still work. worst case you waste some space.